relay-general/src/pii/processor.rs

```diff
@@ -250,12 +230,7 @@ impl<'a> Processor for PiiProcessor<'a> {
         }
 
         // apply rules based on key/path
-        for rule in self.iter_rules(state) {
-            match apply_rule_to_value(meta, rule, state.path().key(), None) {
-                Ok(()) => continue,
-                other => return other,
-            }
-        }
+        self.apply_all_rules(meta, state, None)?;
```

Comment:
_nit_: Directly return the result.

```suggestion
        self.apply_all_rules(meta, state, None)
```